### PR TITLE
Bugfix: division by zero on ratio

### PR DESCRIPTION
**Pull request by fabiob**
Some very small images were reporting a size of zero after downloading, which in turn caused a division by zero when calculating the final ratio.

After adding a new test, it was discovered that it was a filesystem caching issue. A flush operation was added to fix it.

When trying to write the test, I had a hard time debugging the handler as it performed several tasks: parsing the lambda context, coordinating the image downloading and processing, and then writing a valid API Gateway response. So I refactored it into two functions: the `handler` function now only parses the context and writes the response, while the download and image processing jobs are handled by a new `download_and_optimize` function.

### CODE REVIEW

File: imaginex_lambda/handler.py [hunk 1]
```diff
@@ -85,42 +90,61 @@ def handler(event, context):
...
+
+
 if __name__ == '__main__':
     print("Running test...")
 
-    data = json.load(open('example/event-absolute-jfif.json'))
-    handler(data, None)
+    context = {'queryStringParameters':
+               {'q': '40', 'w': '250',
+                'url': 'https://s3.eu-central-1.amazonaws.com/fllite-dev-main/'
```
Comment (sladg) [inline imaginex_lambda/handler.py:148]: Let's change this one to `http://sladg-imaginex.s3.eu-central-1.amazonaws.com`
I will upload some set of example images there

Comment (sladg) [reply imaginex_lambda/handler.py:148]: Aka. `S3_BUCKET_NAME=sladg-imaginex`

**Review by sladg — COMMENTED** [review]
Public S3 created, loads of sample images uploaded.
See: https://sladg-imaginex.s3.eu-central-1.amazonaws.com as it will return list of all objects available.

We can use it for both Absolute URL as well as S3 url's testing.
Credits to https://github.com/ImageGlass/sample-images

File: imaginex_lambda/handler.py (hunk 1 above)
Comment (fabiob) [reply imaginex_lambda/handler.py:148]: I'd rather send a new PR with the new bucket and new tests, is that OK?

Here's a preview: https://github.com/fabiob/imaginex-lambda/commit/259a5a1bfa1984a4cbf1c70fc6db18e02e18707c

Comment (sladg) [reply imaginex_lambda/handler.py:148]: Sounds good!

**Pull request merged**